File: src/main/java/ms/uk/eclipse/commands/tournament/JoinCommand.java
package ms.uk.eclipse.commands.tournament;

import ms.uk.eclipse.PracticePlugin;
import ms.uk.eclipse.core.commands.PlayerCommand;
import ms.uk.eclipse.entity.Profile;
import ms.uk.eclipse.entity.PlayerStatus;
import ms.uk.eclipse.managers.PlayerManager;
import ms.uk.eclipse.managers.TournamentManager;
import ms.uk.eclipse.tournaments.Tournament;
import ms.uk.eclipse.util.messages.ErrorMessages;
import ms.uk.eclipse.util.messages.UsageMessages;

public class JoinCommand extends PlayerCommand {

    final PlayerManager playerManager = PracticePlugin.INSTANCE.getPlayerManager();
    final TournamentManager tournamentManager = PracticePlugin.INSTANCE.getTournamentManager();

    public JoinCommand() {
        super("join");
    }

    @Override
    public void execute(org.bukkit.entity.Player player, String[] args) {
        Profile p = playerManager.getProfile(player);

        if (args.length < 1) {
            p.message(UsageMessages.JOIN);
            return;
        }

        if (p.getPlayerStatus() == PlayerStatus.IN_TOURAMENT) {
            p.message(ErrorMessages.ALREADY_IN_TOURNAMENT);
            return;
        }

        if (p.getPlayerStatus() != PlayerStatus.IN_LOBBY) {
            p.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY);
            return;
        }

        Tournament t = tournamentManager.getTournamentByName(args[0]);

        if (t == null) {
            p.message(ErrorMessages.TOURNAMENT_NOT_EXIST);
            return;
        }

        t.addPlayer(p);
    }
}

File: src/main/kotlin/gg/mineral/practice/commands/contest/JoinCommand.kt
package gg.mineral.practice.commands.contest

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.events.Event
import gg.mineral.practice.managers.ContestManager.getByName
import gg.mineral.practice.tournaments.Tournament
import gg.mineral.practice.util.messages.impl.ErrorMessages

@Command(name = "join")
class JoinCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg name: String) {
        profile.contest?.let {
            return when (it) {
                is Tournament -> profile.message(ErrorMessages.ALREADY_IN_TOURNAMENT)
                is Event -> profile.message(ErrorMessages.ALREADY_IN_EVENT)
                else -> error("Unknown contest type")
            }
        }
        if (profile.playerStatus !== PlayerStatus.IDLE) return profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY)

        val contest = getByName(name)

        if (contest?.addPlayer(profile) == false) profile.message(ErrorMessages.EVENT_TOURNAMENT_NOT_EXIST)
    }
}
